is_cjk condition by @chrissimpkins

felipesanches opened the pull request.

felipesanches commented:
I am refactoring the code a bit, but in general I like the approach.

I think that a good strategy here could consist of a very broad detection of what looks like a CJK font (your `is_cjk` condition) followed by a set of checks to ensure all bits are properly set.

This may sound a bit redundant, but it fits my mindset of (1) a loose / broad detection + (2) finer-grained checking of all details.

felipesanches commented:
This has to do with answering the question "what is the source of truth for determining if a font is actually a CJK font?"

I noticed that `nototools/noto_lint.py` uses the font filename suffix as the source of truth, which is not a general solution to the problem, but instead is a very Noto-specific approach. That's why I decided to use @chrissimpkins' approach instead for the detections of CJK fonts when porting notolint into FontBakery.

felipesanches commented:
(so that in the future some of the noto-specific checks may be migrated into more general profiles such as the `universal` one)

chrissimpkins commented:
@felipesanches have you run the condition on Noto CJK fonts?  Are they properly filtered in this condition check?  I haven't looked yet.

chrissimpkins commented:
For the Noto CJK project, this may be an important requirement: https://github.com/googlefonts/fontbakery/issues/2595

I believe that font collections are part of the releases.

felipesanches commented:
I am happy with this. I simplified the code by deleting the helper functions and doing the bit shifting in-place. I think it is clearer and quicker to grasp at a glance this way.

I also fixed an off-by-one bug: the `range` function in python does not include the final value so we have to plus 1.

m4rc1e commented:
I've been chatting to Cosimo and it sounds like bit manipulating the os2.ulCodePageRanges may not be a good approach. These bits are often set incorrectly so it may be better to analyse the best cmap table instead.

chrissimpkins commented:
> These bits are often set incorrectly so it may be better to analyse the best cmap table instead.

cmap is currently in there as a fallback to the bit flag checks.  If the bit flag checks are removed for CJK, I suggest that we add a test to confirm that those bits *are* properly set when fonts are identified as "CJK".  MS products do not play well with CJK fonts without these properly set.

What a "CJK" font is for the purposes of the bit flag settings is an open question.  Some teams (incl Adobe as I understand) use a very low glyph set inclusion bar to meet the requirement.
